src/main.rs

```diff
@@ -1,3 +1,141 @@
+use clap::{command, Arg};
+use std::process;
+use std::str::FromStr;
+use std::{env, fs};
+use std::path::Path;
+
+enum Mode {
+    Encrypt, // Encryption mode
+    Decrypt, // Decryption mode
+}
+
+impl FromStr for Mode {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Mode, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "encryption" => Ok(Mode::Encrypt),
+            "decryption" => Ok(Mode::Decrypt),
+            _ => Err("invalid mode, must be 'encrypt' or 'decrypt'"),
+        }
+    }
+}
+
+struct Arguments {
+    file_name: String,
+    mode: Mode,
+    key: u8,
+}
+
+impl Arguments {
+    fn new() -> Result<Arguments, &'static str> {
+        let args = command!()
+            .arg(
+                Arg::new("filename")
+                    .short('f')
+                    .long("file")
+                    .required(true)
+            )
+            .arg(
+                Arg::new("mode")
+                    .short('m')
+                    .long("mode")
+                    .required(true)
+            )
+            .arg(
+                Arg::new("key")
+                    .short('k')
+                    .long("key")
+                    .required(true)
+            )
+            .get_matches();
+
+        let file_name: String = args
+            .get_one::<String>("filename")
+            .ok_or("error while getting filename argument")?
+            .clone();
+
+        let mode: Mode = args
+            .get_one::<String>("mode")
+            .ok_or("error while getting mode argument")?
+            .parse()
+            .map_err(|_| "Invalid mode argument")?;
+
+        let key: u8 = match args.get_one::<String>("key") {
+            Some(k) => k
+                .to_string()
+                .parse()
+                .map_err(|_| "error while parsing key, expected u8 number")?,
+            None => return Err("error while get key argument")
+        };
+
+        Ok(Arguments {
+            file_name,
+            mode,
+            key,
+        })
+    }
+}
```

Comment:
```suggestion
#[derive(Parser, Debug)]
#[command(filename, mode, key)]
struct Arguments {
    #[arg(id = "filename", long, short)]
    file_name: String,
    #[arg(long, short)]
    mode: Mode,
    #[arg(long, short)]
    key: u8,
}
```
Фактически обеспечивает такой же функционал